WordPress/Classes/ViewRelated/Post/PostSettings/Views/Excerpt/PostSettingsGenerateExcerptView.swift

```diff
@@ -0,0 +1,459 @@
+
+            Slider(
+                value: Binding(
+                    get: { Double(length.rawValue) },
+                    set: { length = GeneratedContentLength(rawValue: Int($0)) ?? .medium }
+                ),
+                in: 0...Double(GeneratedContentLength.allCases.count - 1),
+                step: 1
+            ) {
+                Text(Strings.lengthSliderAccessibilityLabel)
+            } minimumValueLabel: {
+                Image(systemName: "textformat.size.smaller")
+                    .font(.caption2)
+                    .foregroundStyle(.secondary)
+            } maximumValueLabel: {
+                Image(systemName: "textformat.size.larger")
+                    .font(.caption2)
+                    .foregroundStyle(.secondary)
+            }
+            .tint(AppColor.primary)
+            .onChange(of: length) { _ in
+                didChangeGenerationParameters()
+            }
+        }
+    }
+
+    private var stylePicker: some View {
+        HStack(spacing: 0) {
+            Label {
+                Text(Strings.styleLabel)
+                    .font(.subheadline)
+                    .fontWeight(.medium)
+            } icon: {
+                Image(systemName: "wand.and.rays")
+                    .font(.caption)
+                    .foregroundStyle(.secondary)
+            }
+
+            Spacer(minLength: 8)
+
+            Picker(Strings.stylePickerAccessibilityLabel, selection: $style) {
+                ForEach(GenerationStyle.allCases, id: \.self) { style in
+                    Text(style.displayName)
+                        .tag(style)
+                }
+            }
+            .pickerStyle(.menu)
+            .tint(Color.primary)
+            .onChange(of: style) { _ in
+                didChangeGenerationParameters()
+            }
+        }
+    }
+
+    // MARK: - Generation
+
+    private func cancelAllTasks() {
+        generationTask?.cancel()
+        debounceTask?.cancel()
+    }
+
+    private func generateExcerpts() {
+        generationTask?.cancel()
+
+        excerpts = []
+        isGenerating = true
+        isFirstResult = true
+        error = nil
+
+        generationTask = Task {
+            do {
+                try await startGeneration()
+            } catch {
+                if !Task.isCancelled {
+                    self.error = error
+                }
+            }
+            if !Task.isCancelled {
+                isGenerating = false
+            }
+        }
+    }
+
+    private func didChangeGenerationParameters() {
+        debounceTask?.cancel()
+        debounceTask = Task {
+            try? await Task.sleep(for: .milliseconds(330))
+            guard !Task.isCancelled else { return }
+            generateExcerpts()
+        }
+    }
+
+    private func startGeneration() async throws {
+        let session = LanguageModelSession()
+        let prompt = LanguageModelHelper.makeGenerateExcerptPrompt(
+            content: postContent,
```

Comment:
Is it possible to use one single `LanguageModelSession`, with `postContent` (which won't be changed during the session) loaded into it? I think that should speed up the generation process, because it does not need to process the same `postContent` again and again in the `startGeneration` function.

Reply:
My understanding is `LanguageModelSession` should be reused only if you want to keep context between interactions. I don't think this is what you want when changing the criteria. I did just add a "Suggest More Options" button to generate more results where I reuse `LanguageModelSession`. From my testing, even running a simple prompt like "generate more results" in the same session still takes roughly the same amount of time as starting a new session.

https://github.com/user-attachments/assets/3bff2050-5a2a-4b1d-8d11-874bbe793d37